### PR TITLE
Bug 1238870 - Backspace key for back function

### PR DESCRIPTION
**Pull request by rickychien**
https://bugzilla.mozilla.org/show_bug.cgi?id=1238870

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=aa75c1582d6462c8b82f0aaedfeff825452db4f3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b2ae211ae0890863bfbc8d4ddb052e21d253740a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1dc1e0c1058a21d7aaa29b1efe096d9336f9d6af

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b2ae211ae0890863bfbc8d4ddb052e21d253740a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1489fc774f84ed06250196e79b1fb953fc646927

File: shared/js/smart-screen/key_navigation_adapter.js [hunk 1]
```diff
@@ -84,8 +84,6 @@
           return 'left';
         case KeyEvent.DOM_VK_RETURN:
           return 'enter';
-        case KeyEvent.DOM_VK_ESCAPE:
```
Comment (luke-chang) [inline shared/js/smart-screen/key_navigation_adapter.js]: I suggest keeping this part as is.

File: tv_apps/app-deck/bower_components/smart-dialog/script.js [hunk 2]
```diff
@@ -96,9 +96,9 @@ window.SmartDialog = (function(win) {
         }
         break;
       case 'keyup':
-        // close dialog when ESC is clicked
+        // close dialog when Backspace is clicked
```
Comment (luke-chang) [inline tv_apps/app-deck/bower_components/smart-dialog/script.js]: Remember to modify the corresponding smart components as well.

File: tv_apps/smart-home/bower_components/smart-dialog/script.js [hunk 3]
```diff
@@ -98,7 +98,7 @@ window.SmartDialog = (function(win) {
       case 'keyup':
         // close dialog when ESC is clicked
         if (this.getAttribute('esc-close') != 'false' &&
-            (evt.keyCode === 27 || evt.key === 'Esc') &&
+            evt.keyCode === KeyEvent.DOM_VK_BACK_SPACE &&
```
Comment (luke-chang) [inline tv_apps/smart-home/bower_components/smart-dialog/script.js]: Remember to modify the corresponding smart component as well.

File: tv_apps/smart-system/bower_components/smart-dialog/script.js [hunk 4]
```diff
@@ -98,7 +98,7 @@ window.SmartDialog = (function(win) {
       case 'keyup':
         // close dialog when ESC is clicked
         if (this.getAttribute('esc-close') != 'false' &&
-            (evt.keyCode === 27 || evt.key === 'Esc') &&
+            evt.keyCode === KeyEvent.DOM_VK_BACK_SPACE &&
```
Comment (luke-chang) [inline tv_apps/smart-system/bower_components/smart-dialog/script.js]: Remember to modify the corresponding smart component as well.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=445ffed6c185960e2f92a32b022b45b1ca38b829

**Pull request merged**